content/blog/2020/05/2020-05-18-read-only-jenkins-announcement.adoc

```diff
@@ -0,0 +1,126 @@
+The problem with this is you don’t know when new plugin versions are available and in order to see what other configuration options are available to a plugin you currently need the 'Administer' permission.
+
+Allowing non administrator users to debug issues with their builds easier, i.e. given a 'Jenkins' error message in a build the user can check:
+
+1. which plugins are installed, 
+2. the version of the plugin
+
+This can allow the user to solve their issue themselves and makes it easier for the user to report an issue with a plugin directly to the maintainers.
+
+== What can I expect
+
+All built in UI controls have been adapted to clearly distinguish between
+an editable control and a control you don't have permission to edit:
+
+Editable:
+
+image:/images/post-images/2020/05-read-only-jenkins-announcement/build-discarder-edit.png[]
+
+Non editable:
+
+image:/images/post-images/2020/05-read-only-jenkins-announcement/build-discarder-read.png[]
+
+
+_Note: there are other controls such as in the credentials and pipeline plugins that have
+not been updated yet._
+
+Action buttons, (Such as 'Save' and 'Apply') have been hidden in most cases.
+
+There is more work on this needed, some plugins need support added and certain controls 
+could have some improvements done to render better.
+
+== How can I use it?
+
+These permissions are currently available in beta and for now disabled by default,
+you can enable them by installing the plugin:extended-read-permission[Extended read permission plugin].
+
+Then you will need to add the following permissions to a user / group depending on your usecase:
+
+* Overall/SystemRead
+* Job/ExtendedRead
+* Agent/ExtendedRead
+
+_Note: You will need to set the Overall/Read and Job/Read permissions as well, you might
+want to consider creating a role containing the required permissions._
+
+Here is an example using the plugin:configuration-as-code[Configuration as Code plugin] and the plugin:folder-auth[Folder-based Authorization Strategy plugin]:
+
+[source,yaml]
+----
+jenkins:
+  authorizationStrategy:
+    folderBased:
+      globalRoles:
+        - name: "admin"
+          permissions:
+            - id: "Overall/Administer"
+          sids:
+            - "admin"
+        - name: "global read"
+          permissions:
+            - id: "Agent/ExtendedRead"
+            - id: "Overall/SystemRead"
+            - id: "Overall/Read"
+            - id: "Job/Read"
+            - id: "Job/ExtendedRead"
+          sids:
+            - "reader" 
+----
+
+== I can't see a configuration that I think should be allowed
+
+Most of Jenkins itself has been updated to support read-only Jenkins, but not very many plugins,
+please create an enhancement issue on the plugins issue tracker, if they use Jira then
+you can add it to the jira:JENKINS-12548[] epic.
+
+== How do I update my plugin to support it
+
+See the link:/doc/developer/views/read-only/[Read only view] section of the developer documentation.
+
+== What's next
+
```

Comment:
```suggestion

In this release we introduce a foundation feature which is already supported in all key Jenkins core controls and in some plugins.
There are many plugins which contribute to Global configurations and diagnostics which still need to be adapted to support the new mode.
We will keep working on this feature and its adoption so that the next LTS baseline in September provides a full-fledged user experience for Jenkins admins. 

```